packages/docprovider/src/tokens.ts

```diff
@@ -35,6 +35,11 @@ export interface IDocumentProvider {
    */
   releaseLock(lock: number): void;
 
+  /**
+   * Returns a Promise that resolves when renaming is ackownledged.
+   */
+  readonly renameAck: Promise<boolean>;
```

Comment:
This addition to the new `IDocumentProvider` will probably break downstream implementations if this gets backported to 3.3.x and they update to the new packages, since this field is not optional?

For example JupyterLite implements the interface here: https://github.com/jupyterlite/jupyterlite/blob/41da1cb7ba64708284a07ac381d296466d0f0fdd/packages/application-extension/src/index.tsx#L66

Reply:
We could say this is fine, if we consider there are not that many custom implementations out there yet.

Reply:
I think we could make this field optional. It is used by the JupyterLab implementation in this PR but I guess other implementations would not care to know if/when renaming actually completes.

Reply:
We could make it optional with a `TODO` to make it required for the 4.0 version?